tasks/agent.py

```diff
@@ -666,7 +667,22 @@ def bundle_install_omnibus(ctx, gem_path=None, env=None):
         cmd = "bundle install"
         if gem_path:
             cmd += f" --path {gem_path}"
-        ctx.run(cmd, env=env)
+
+        for trial in range(MAX_TRY_BUNDLE_INSTALL):
+            res = ctx.run(cmd, env=env, warn=True)
+            if res.ok:
+                return
+            if not should_retry_bundle_install(res):
+                return
+            print(f"Retrying bundle install, attempt {trial + 1}/{MAX_TRY_BUNDLE_INSTALL}")
+
+
+def should_retry_bundle_install(res):
+    # We sometimes get a Net::HTTPNotFound error when fetching the
+    # license-scout gem. This is a transient error, so we retry the bundle install
+    if "Net::HTTPNotFound:" in res.stderr:
+        return True
+    return False
```

Comment:
```suggestion
    return "Net::HTTPNotFound:" in res.stderr
```

Reply:
I wrote it like that to be able to easily add new `if` to be able to retry on other error. If that's fine for you I will let it as is